src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,277 @@
+        {
+            Guard.NotNullOrEmpty(lockToken, nameof(lockToken));
+
+            await _messageReceiver.DeadLetterAsync(lockToken, reason, errorDescription);
+        }
+
+        private MessageHandlerOptions DetermineMessageHandlerOptions(AzureServiceBusMessagePumpSettings messagePumpSettings)
+        {
+            var messageHandlerOptions = new MessageHandlerOptions(HandleReceivedExceptionAsync);
+            if (messagePumpSettings.Options != null)
+            {
+                // Assign the configured defaults
+                messageHandlerOptions.AutoComplete = messagePumpSettings.Options.AutoComplete;
+                messageHandlerOptions.MaxConcurrentCalls = messagePumpSettings.Options.MaxConcurrentCalls ?? messageHandlerOptions.MaxConcurrentCalls;
+
+                Logger.LogInformation("Message pump options were configured instead of Azure Service Bus defaults.");
+            }
+            else
+            {
+                Logger.LogWarning("No message pump options were configured, using Azure Service Bus defaults instead.");
+            }
+
+            return messageHandlerOptions;
+        }
+
+        private async Task<MessageReceiver> CreateMessageReceiverAsync(AzureServiceBusMessagePumpSettings messagePumpSettings)
+        {
+            var rawConnectionString = await messagePumpSettings.GetConnectionStringAsync();
+            var serviceBusConnectionStringBuilder = new ServiceBusConnectionStringBuilder(rawConnectionString);
+
+            MessageReceiver messageReceiver;
+            if (string.IsNullOrWhiteSpace(serviceBusConnectionStringBuilder.EntityPath))
+            {
+                // Connection string doesn't include the entity so we're using the message pump settings
+                if (string.IsNullOrWhiteSpace(messagePumpSettings.EntityName))
+                {
+                    throw new ArgumentNullException(nameof(messagePumpSettings.EntityName), "No entity name was specified while the connection string is scoped to the namespace");
+                }
+
+                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, messagePumpSettings.EntityName, messagePumpSettings.SubscriptionName);
+            }
+            else
+            {
+                // Connection string includes the entity so we're using that instead of the message pump settings
+                messageReceiver = CreateReceiver(serviceBusConnectionStringBuilder, serviceBusConnectionStringBuilder.EntityPath, messagePumpSettings.SubscriptionName);
+            }
+
+            Namespace = messageReceiver.ServiceBusConnection?.Endpoint?.Host;
+
+            ConfigurePlugins();
+
+            return messageReceiver;
+        }
+
+        private MessageReceiver CreateReceiver(ServiceBusConnectionStringBuilder serviceBusConnectionStringBuilder, string entityName, string subscriptionName)
+        {
+            EntityPath = entityName;
```

Comment:
We cannot since we depend on the connection string which is an async operation, unless async ctors are already there ofcourse, but I don't think so?

`var rawConnectionString = await messagePumpSettings.GetConnectionStringAsync();`

Reply:
Just to be clear, fully agree but I was forced to do this

Reply:
Hmm, we don't need that `async` call, do we? Only `messagePumpSettings.EntityName` and `messagePumpSettings.SubscriptionName`?

Reply:
Yes we do as the entity path can be part of the connection string.

This should also be our recommended approach as it scopes the authentication to only that entity and not the whole namespace